lib/inspec/profile.rb

```diff
@@ -89,7 +89,7 @@ def initialize(source_reader, options = {})
       @writable = options[:writable] || false
       @profile_id = options[:id]
       @cache = options[:cache] || Cache.new
-      @backend = options[:backend] || Inspec::Backend.create(options)
+      @backend = options[:backend] || Inspec::Backend.create(options.select { |k, _| k != 'target' })
```

Comment:
Agreed that additional testing is always a good thing. I don't agree that that's the "real issue." We have a situation in the codebase where `A` is creating an options hash, `B` is passing it to `C`, but not all the options are needed by `C` and, when they are all included, we break. So even if "all the ssh and winrm options are just passed through," we're not doing a good job with the separation. Testing will catch it, but the lack of testing is not the cause here.